pkg/charts/utils.go

```diff
@@ -0,0 +1,187 @@
+)
+
+var defaultHubbleConfig = hubbleConfig{
+	UI: hubbleUI{
+		Enabled: false,
+	},
+	Metrics: hubbleMetrics{
+		Enabled: []string{"dns", "drop", "tcp", "flow", "port-distribution", "icmp", "http"},
+	},
+}
+var defaultCiliumConfig = ciliumConfig{
+	Agent: agent{
+		Enabled:              true,
+		SleepAfterInit:       false,
+		KeepDeprecatedLabels: false,
+	},
+	Config: config{
+		Enabled: true,
+	},
+	Operator: operator{
+		Enabled: true,
+	},
+	Preflight: preflight{
+		Enabled:         false,
+		TofqdnsPreCache: "",
+	},
+}
+
+var defaultGlobalConfig = globalConfig{
+	IdentityAllocationMode: ciliumv1alpha1.CRD,
+	Tunnel:                 ciliumv1alpha1.VXLan,
+	KubeProxyReplacement:   ciliumv1alpha1.Probe,
+	Etcd: etcd{
+		Enabled:       true,
+		Managed:       true,
+		ClusterDomain: "cluster.local",
+		Endpoints:     "",
+		SSL:           false,
+	},
+	Ipv4: ipv4{
+		Enabled: true,
+	},
+	Ipv6: ipv6{
+		Enabled: false,
+	},
+	Debug: debug{
+		Enabled: false,
+	},
+	Prometheus: prometheus{
+		Enabled: false,
+		Port:    9090,
+		ServiceMonitor: serviceMonitor{
+			Enabled: false,
+		},
+	},
+	OperatorPrometheus: operatorPrometheus{
+		Enabled: false,
+		Port:    6942,
+	},
+	Psp: psp{
+		Enabled: true,
+	},
+	Images: map[string]string{
+		cilium.CiliumAgentImageName:        imagevector.CiliumImage(),
+		cilium.CiliumOperatorImageName:     imagevector.CiliumOperatorImage(),
+		cilium.CiliumETCDOperatorImageName: imagevector.CiliumEtcdOperatorImage(),
+		cilium.CiliumNodeInitImageName:     imagevector.CiliumNodeInitImage(),
+		cilium.CiliumPreflightImageName:    imagevector.CiliumPreflightImage(),
+
+		cilium.HubbleImageName:   imagevector.CiliumHubbleImage(),
+		cilium.HubbleUIImageName: imagevector.CiliumHubbleUIImage(),
+	},
+}
+
+func newGlobalConfig() globalConfig {
+	return defaultGlobalConfig
+}
+
+func newHubbleConfig() hubbleConfig {
+	return defaultHubbleConfig
+}
+
+func newCiliumConfig() ciliumConfig {
+	return defaultCiliumConfig
+}
+
+func toMap(c interface{}) (map[string]interface{}, error) {
+	bytes, err := json.Marshal(c)
+	if err != nil {
+		return nil, fmt.Errorf("could not marshal cilium config: %v", err)
+	}
+	var configMap map[string]interface{}
+	err = json.Unmarshal(bytes, &configMap)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal cilium config: %v", err)
+	}
+	return configMap, nil
+}
+
+// ComputeCiliumChartValues computes the values for the cilium chart.
+func ComputeCiliumChartValues(network *extensionsv1alpha1.Network, config *ciliumv1alpha1.NetworkConfig) (map[string]interface{}, error) {
+	ciliumConfig, globalConfig, hubbleConfig, err := generateChartValues(config)
+	if err != nil {
+		return nil, fmt.Errorf("error when generating config values %v", err)
+	}
+
+	ciliumConfigmap, err := toMap(ciliumConfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert cilium config: %v", err)
+	}
+
+	globalConfigmap, err := toMap(globalConfig)
+
+	ciliumChartValues := map[string]interface{}{
```

Comment:
Gardener is using PodSecurityPolicy by default. it's disabled in the values file. Enable it.

Reply:
its overriden in the deault values to true.